src/raw.js

```diff
@@ -136,6 +140,10 @@ function replaceKeyBindings(raw) {
     return full.replace(key, '?')
   })
 
+  emptyBindings.map(function(binding) {
+    sql = sql.replace(new RegExp(` ${binding}|${binding} |${binding}`, 'g'), '');
```

Comment:
@elhigu I agree. This function should only replace bindings when it's possible. My proposal is this:

If a binding is present in the query, but is undefined in the bindings object, then simply leave the binding in the sql untouched. That will in itself throw a query error, and thanks to not replacing the binding, it will be obvious what went wrong via the error message. In other words, delete the code that deals with `emptyBindings`.

```javascript
knex.raw('SELECT :val', {val: 1}); //SELECT 1

knex.raw('SELECT :val', {val: void 0}); //SELECT :val -- Throws error
```

You agree and good with merging after this? I expect a few tests to change to accomodate for removing that code, like the references above. I also plan to add this to the docs.